en/tutorials/chat-app.md

```diff
@@ -0,0 +1,400 @@
+Before we get started with the JavaScript, we'll add some unique CSS to our **www/assets/main.css** file:
+
+```css
+.no-resize { resize: none; }
+
+.flex-right { 
+  -webkit-justify-content: flex-end; 
+  -moz-justify-content: flex-end; 
+  -ms-justify-content: flex-end; 
+  justify-content: flex-end; 
+}
+
+.user-avatar::before {
+  color: black;
+  content: "Click to change avatar ->";
+  display: inline-block;
+  font-size: 1rem;
+  height: 1rem;
+  opacity: 0;
+  position: absolute;
+  right: 130%;
+  transition: opacity 0.3s ease-out;
+  top: 25%;
+  width: 300%;
+}
+
+.user-avatar:hover::before {
+  opacity: 1;
+}
+
+.user-avatar:hover {
+  cursor: pointer;
+}
+
+.input-container {
+  z-index: 21;
+}
+```
+
+Now we can finally open up **www/assets/js/main.js** and start adding some functionality to our little chat app. That file should already have the one line needed to initialize a new Hoodie class:
+
+```js 
+// initialize Hoodie
+var hoodie  = new Hoodie();
+```
+
+From there, we can create some dedicated Stores for quickly saving and finding our messages and avatars:
+
+```js
+// create a scoped 'message' store for easier re-use
+var messageStore = hoodie.store('message');
+
+// create a scoped 'avatar' store for easier re-use
+var avatarStore = hoodie.store('avatar');
+```
+
+Another part of our app setup will be selecting all the DOM elements we'll be interacting with to make the app work using the custom **data-action** attribute we gave them earlier:
+
+```js
+// Select the chat form
+var chatForm = $('[data-action="chat-input"]');
+
+// Select the textarea from the chat form
+var chatBox = $('[data-action="send-message"]');
+
+// Select the chat stream area
+var chatStream = $('[data-action="chat-stream"]');
+```
+
+As for DOM interaction, we can begin by listening to the **chatForm** for **submit** events and send the message to our message store when that happens:
+
+```js
+// Setup submit event listener on chat form
+chatForm.on('submit', sendMessage);
+
+// create sendMessage function
+function sendMessage(e) {
+  e.preventDefault();
+
+  // check for content in the chatBox,
+  // if there is content,
+  // then assign it to a variable
+  // else, return false to cancel function
+  var messageContent = chatBox.val().trim();
+
+  if ( messageContent.length < 1 ) { return false; }
+
+  // create a new message model
+  var message = new messageModel(messageContent);
+
+  // using the global messageStore, add this message object and publish it to the global store.
+  messageStore.add(message).publish();
+
+  trigger an immediate sync with the server for quicker updates
+  hoodie.remote.push();
+
+  // Dont't forget to clear out the charBox
+  chatBox.val('');
+}
+```
+
+You may have noticed the **new messageModel(messageContent)**; we define this **messageModel** just after defining the **sendMessage** function, in order to abstract some of the logic that goes into making our message object we're saving to the **messageStore**:
```

Comment:
Ah, here is the `messageModel`, maybe worth making a note further up that it comes later.